File: DiseoAppIMC/app/src/main/java/com/example/diseoappimc/MainActivity.java
package com.example.diseoappimc;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    //Declaracion de componentes a utlizar
    private EditText etxPeso, etxEstatura;
    private TextView txvIMC;
    private Button btnCalcular;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //Asociacion de objetos; controlador con vista
        etxPeso = (EditText)findViewById(R.id.etxPeso);
        etxEstatura = (EditText)findViewById(R.id.etxEstatura);
        btnCalcular = (Button) findViewById(R.id.btnCalcular);
        txvIMC = (TextView) findViewById(R.id.txvIMC);
        ImageView img = (ImageView)findViewById(R.id.imageView);

        //Manejo de evento en Java
        btnCalcular.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                double peso = Double.parseDouble(etxPeso.getText().toString());
                double estatura = Double.parseDouble(etxEstatura.getText().toString());
                double imc = peso / (estatura * estatura);
                String mensaje = "";
                if (imc>0 && imc<19) {
                    mensaje = "Bajo peso";
                } else if (imc>19 && imc<25) {
                    mensaje = "Peso normal";
                } else if (imc>25 && imc<50) {
                    mensaje ="Sobrepeso";
                } else {
                    mensaje = "Dato inválido";
                }
                txvIMC.setText("IMC = " + imc + "\n" + mensaje);
            }
        });
    }
}

File: AppLeyOhm/app/src/main/java/com/example/appleyohm/MainActivity.kt
package com.example.appleyohm

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        //Manejo del evento en Kotlin con un escucha
        btnCalcular.setOnClickListener() {
            //Obtencion de valores de la vista
            var voltaje : Double = etxVoltaje.text.toString().toDouble()
            var resistencia : Double = etxResistencia.text.toString().toDouble()
            var corrriente : Double = etxCorriente.text.toString().toDouble()
            var resultado : Double? //Valor desconocido
            var mensaje : String = ""

            if (rbtVoltaje.isChecked) {
                //Se calcula el Voltaje V = I x R
                resultado = corrriente * resistencia
                etxVoltaje?.setText("vda")
                mensaje = "Se calculo el Voltaje"
            } else if (rbtCorriente.isChecked) {
                //Se calcula la corriente I = V / R
                resultado = valorUno / valorDos
                txvResultado.text = "La corriente es de ${resultado.toString()} amperios"
                mensaje = "Se calculo la corriente"
            } else if (rbtResistencia.isChecked) {
                //Se calcula la resistencia R = V / I
                resultado = valorUno / valorDos
                txvResultado.text = "La resistencia es de ${resultado.toString()} ohms"
                mensaje = "Se calculo la resistencia"
            } else {
                mensaje = "No se realizo ningun calulo"
            }
            Toast.makeText(this, mensaje, Toast.LENGTH_LONG).show()
        }
    }
}
